src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -13,6 +13,11 @@ public class StringFunctions {
     private static final char SPACE = ' ';
 
     public static final String reverseWithStringConcat(String string) {
+        if (string == null || string.isEmpty()){
```

Comment:
Added check to return empty string or NULL to prevent exceptions during the code.